### PR TITLE
PreformattedText component

### PR DESCRIPTION
**Pull request by KavithaKurella**

### CODE REVIEW

File: stories/PreformattedText/Documentation.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+A PreformattedText component ready to use in your form.  
+PreformattedText is UI/UX agnostic so you need to provide your style to have the look and feel you prefer.  
+When you import the PreformattedText component without providing any className or style associated will looks as following:
+
+<Canvas of={PreformattedTextStories.Unstyled} />
+
+A more complex example with all the possible properties is:
+
+```js
+<PreformattedText class="" id="user-defined-id">
```
Comment (daniele-zurico) [inline stories/PreformattedText/Documentation.mdx:17]: please specify all the possible properties in this example

File: stories/PreformattedText/ClassBased.stories.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  tags: ['autodocs'],
+};
+
+export const Basic = {
+  name: 'Basic',
+  args: {
+    value:
+      'Text in a pre element is displayed in a fixed-width font, and it preserves both      spaces and line breaks.',
+    className: '',
```
Comment (daniele-zurico) [inline stories/PreformattedText/ClassBased.stories.js:23]: if it's classBased means that you need to specify a className here, define a css file in this folder and import it

File: src/preformattedText/PreformattedText.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+type preformattedTextProps = {
+  /**
+   * A CSS class for styling the preformatted text
+   */
+  className?: string;
+  /**
+   * The value of the preformatted text
+   */
+  value?: string;
```
Comment (daniele-zurico) [inline src/preformattedText/PreformattedText.tsx:12]: I think this is mandatory

File: stories/PreformattedText/Documentation.mdx (hunk 1 above)
Comment (KavithaKurella) [reply stories/PreformattedText/Documentation.mdx]: I have solved this.

File: stories/PreformattedText/ClassBased.stories.js (hunk 2 above)
Comment (KavithaKurella) [reply stories/PreformattedText/ClassBased.stories.js]: I have solved this.

File: stories/govUkStyle.css [hunk 4]
```diff
@@ -7,6 +7,12 @@
   display: block;
   clear: both;
 }
+.customPreformattedText {
```
Comment (daniele-zurico) [inline stories/govUkStyle.css:10]: I probably explained it before but let me re-iterate. You use govUkStyle.css if the style you're creating is related to govUk if instead is not for that you create a new file in the folder where you working on called style.css and you include in the folder and you import here.
Example:
stories/PreformattedText/style.css

and then here you do:
import "./style.css"

File: stories/PreformattedText/Documentation.mdx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+PreformattedText is UI/UX agnostic so you need to provide your style to have the look and feel you prefer.  
+When you import the PreformattedText component without providing any className or style associated will looks as following:
+
+<Canvas of={PreformattedTextStories.Unstyled} />
+
+A more complex example with all the possible properties is:
+
+```js
+<PreformattedText
+  class="customPreformattedText"
```
Comment (daniele-zurico) [inline stories/PreformattedText/Documentation.mdx:18]: have you defined a property called `class` in your component? You need to put the properties that you defined there

File: src/preformattedText/PreformattedText.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+   */
+  value: string;
+  /**
+   * Additional props/attributes
+   */
+  props?: React.HTMLAttributes<HTMLPreElement>;
+  /**
+   * An optional id for the preformatted text element
+   */
+  id?: string;
```
Comment (daniele-zurico) [inline src/preformattedText/PreformattedText.tsx:20]: I don't think this property is present in the ticket

File: src/preformattedText/__test__/PreformattedText.test.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  });
+  it('should have the embedded className "dcx-pre"', () => {
+    render(<PreformattedText value="Text in a pre element..." />);
+    expect(screen.getByText('Text in a pre element...')).toHaveClass('dcx-pre');
+  });
+  it('should apply additional classes for shared styling', () => {
+    render(
+      <PreformattedText
+        className="additional-class"
+        value="
```
Comment (daniele-zurico) [inline src/preformattedText/__test__/PreformattedText.test.tsx:19]: put on the same line please... why the content of value is on a new line?

File: stories/PreformattedText/Documentation.mdx (hunk 5 above)
Comment (KavithaKurella) [reply stories/PreformattedText/Documentation.mdx:18]: Yeah I changed it to className which is defined in the component.

File: src/preformattedText/PreformattedText.tsx (hunk 6 above)
Comment (KavithaKurella) [reply src/preformattedText/PreformattedText.tsx:20]: I removed this id property. and also the other things which are related to this.

File: src/preformattedText/__test__/PreformattedText.test.tsx (hunk 7 above)
Comment (KavithaKurella) [reply src/preformattedText/__test__/PreformattedText.test.tsx:19]: I edited this.

File: src/preformattedText/__test__/PreformattedText.test.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+      />
+    );
+    expect(container.firstChild).toHaveAttribute('id', 'user_defined_id');
+  });
+  it('should apply the provided title attribute', () => {
+    const title = 'Preformatted Text';
+    render(<PreformattedText value="Text in a pre element..." title={title} />);
+    const preElement = screen.getByText('Text in a pre element...');
+    expect(preElement).toHaveAttribute('title', title);
+  });
```
[the remark this answers is not in the document]
Comment (KavithaKurella) [reply src/preformattedText/__test__/PreformattedText.test.tsx:55]: I removed this.

File: src/preformattedText/PreformattedText.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  /**
+   * An optional title for the preformatted text element
+   */
+  title?: string;
```
[the remark this answers is not in the document]
Comment (KavithaKurella) [reply src/preformattedText/PreformattedText.tsx:24]: I removed this.

File: stories/govUkStyle.css (hunk 4 above)
Comment (KavithaKurella) [reply stories/govUkStyle.css]: solved this

File: .vscode/settings.json [hunk 10]
```diff
@@ -1,14 +1,11 @@
 {
-  "window.zoomLevel": 0,
```
[the remark this answers is not in the document]
Comment (KavithaKurella) [reply .vscode/settings.json:2]: done

[the remark this answers is not in the document]
Comment (daniele-zurico) [reply .vscode/settings.json:2]: it's still there

[the remark this answers is not in the document]
Comment (daniele-zurico) [reply .vscode/settings.json:2]: And you also have conflicts in your PR

[the remark this answers is not in the document]
Comment (KavithaKurella) [reply .vscode/settings.json]: I have solved.

File: stories/PreformattedText/Documentation.mdx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+<Canvas of={PreformattedTextStories.Unstyled} />
+
+A more complex example with all the possible properties is:
+
+```js
+<PreformattedText
+  className="customPreformattedText"
+  props={{ style: { color: red } }}
+>
+  Text in a pre element is displayed in a fixed-width font, and it preserves
```
Comment (daniele-zurico) [inline stories/PreformattedText/Documentation.mdx:21]: This is really how you'll use the component? I don't think the documentation is correct

File: .vscode/settings.json [hunk 12]
```diff
@@ -4,11 +4,8 @@
     "editor.defaultFormatter": "esbenp.prettier-vscode"
   },
   "editor.formatOnSave": true,
-  "stylelint.validate": [
```
Comment (daniele-zurico) [inline .vscode/settings.json:7]: not sure why you continue to do not reset this file on how it was

File: stories/PreformattedText/Documentation.mdx (hunk 11 above)
Comment (KavithaKurella) [reply stories/PreformattedText/Documentation.mdx:18]: solved.

File: .vscode/settings.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (KavithaKurella) [reply .vscode/settings.json]: solved.

File: .vscode/settings.json (hunk 12 above)
Comment (KavithaKurella) [reply .vscode/settings.json]: solved.

**Review by daniele-zurico — APPROVED** [review]

**Review by Ibabalola — APPROVED** [review]

**Pull request merged**